blocks/editable/index.js

```diff
@@ -131,10 +131,18 @@ export default class Editable extends wp.element.Component {
 		}
 
 		const content = this.getContent();
+		const collapsed = this.editor.selection.isCollapsed();
 
 		this.setState( {
 			empty: ! content || ! content.length,
 		} );
+
+		if ( this.props.focus.collapsed !== collapsed ) {
```

Comment:
If somehow the editor is active and the block is not selected, then yes. Maybe `getBlockFocus` should return an empty object in that case? Or not? Cc @youknowriad.